repos/base-hw/src/bootstrap/spec/zynq/platform.cc

```diff
@@ -23,11 +23,27 @@ Bootstrap::Platform::Board::Board()
   late_ram_regions(Memory_region { RAM_0_BASE, 0x1000 }),
   core_mmio(Memory_region { CORTEX_A9_PRIVATE_MEM_BASE,
                             CORTEX_A9_PRIVATE_MEM_SIZE },
-            Memory_region { UART_0_MMIO_BASE,
+            Memory_region { UART_BASE,
                             UART_SIZE },
             Memory_region { PL310_MMIO_BASE,
                             PL310_MMIO_SIZE }) { }
 
 
 bool Bootstrap::Cpu::errata(Bootstrap::Cpu::Errata err) {
 	return false; }
+
+void Bootstrap::Cpu::wake_up_all_cpus(void* ip) {
+	struct Wakeup_generator : Genode::Mmio
+	{
+		struct Core1_boot_addr : Register<0x0, 32> { };
+
+		Wakeup_generator(void * const ip) : Mmio(CORE1_ENTRY)
+		{
+			write<Core1_boot_addr>((addr_t)ip);
+		}
+	};
+
+	Wakeup_generator wgen(ip);
```

Comment:
The whole Wakeup_generator might be replaced by a one-liner like ``*(void**)CORE1_ENTRY = ip;``. Sub-byte structures (for which Mmio is made) aren't used here anyway. So, it would reduce the LOC.

Reply:
This is also a copy-paste thing. I took this construct from the pandaboard code.

Reply:
I see, on the pandaboard, we use 'Mmio' because of a bit that must be set additionally to the IP. It's only a suggestion though.